### PR TITLE
[MPS] Migrate log_sigmoid forward/backward from MPSGraph to Metal

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.14.0) (oldest at bottom):
* __->__ #187228

Replace the MPSGraph-based `log_sigmoid_forward_*_mps` / `log_sigmoid_backward_mps*` with native Metal kernels in `ActivationKernel.metal` using unary iterator.
Forward  numerically stable formula is $min(0, x) - \log1p(e^{-|x|})$. Backward uses the
observation that $\frac{d}{dx} log(sigmoid(x)) = sigmoid(-x)$, which lets us
write the gradient as `(in<0) ? (1-t) : t` with $t = \frac{z}{1+z}$, $z = e~{-|in|}$.

Replace fail in LogSigmoid float16 OpInfo `test_forward` with xfail to a `toleranceOverride(atol=5e-3, rtol=1e-3)`. 

Microbenchmark (us per call, A/B from the same script with only the impl swapped):

```
shape|dtype                      fwd mpsg   fwd metal  fwd_x   bwd mpsg  bwd metal  bwd_x
1024|float32                        10.68       6.61   1.62x      30.00      15.94   1.88x
1024|float16                        10.85       6.77   1.60x      27.46      17.95   1.53x
1024|bfloat16                       11.83       6.70   1.77x      26.64      14.22   1.87x
1024x1024|float32                   36.35      20.64   1.76x      30.05      41.84   0.72x
1024x1024|float16                   14.68      11.92   1.23x      29.53      27.60   1.07x
8x1024x1024|float32                267.45     252.40   1.06x     378.48     380.34   1.00x
32x1024x1024|float32              1012.71     987.62   1.03x    1499.97    1481.34   1.01x
```

Co-Authored-By: Claude Opus 4.7 <noreply@anthropic.com>

### CODE REVIEW

File: torch/testing/_internal/common_modules.py [hunk 1]
```diff
@@ -4864,9 +4864,16 @@ def module_error_inputs_torch_nn_Pad3d(module_info, device, dtype, requires_grad
...
                module_inputs_func=module_inputs_torch_nn_LogSigmoid,
-               skips=(
-                   # See #119108: tolerance issue
-                   DecorateInfo(unittest.expectedFailure, "TestModule", "test_forward", device_type='mps', dtypes=[torch.float16]),)
+               decorators=(
+                   # The test's reference_fn is `i.sigmoid().log()`, which in fp16
+                   # loses precision as sigmoid(x) saturates near 1 for moderate x.
+                   # log_sigmoid uses the stable `min(0,x) - log1p(exp(-|x|))`
+                   # in fp32, so it can differ from the naive reference by up to
+                   # ~1 ULP of sigmoid(x) (~4e-3 for x in fp16's mid-range).
```
Comment (Skylion007) [inline torch/testing/_internal/common_modules.py:4872]: No way to improve this with precise::math stuff?

File: aten/src/ATen/native/mps/kernels/ActivationKernel.metal [hunk 2]
```diff
@@ -323,3 +323,33 @@ REGISTER_BINARY_OP(sigmoid_backward, half, half);
...
+
+struct log_sigmoid_backward_functor {
+  template <typename T>
+  inline T operator()(const T self, const T grad_output) {
+    // d/dx log(sigmoid(x)) = 1 - sigmoid(x) = sigmoid(-x); compute it stably
+    // via z = exp(-|x|), splitting on sign(x) to avoid overflow.
+    const float in = float(self);
+    const float z = ::metal::precise::exp(-::metal::abs(in));
+    const float t = z / (1.0f + z);
+    return static_cast<T>(float(grad_output) * (in < 0.0f ? 1.0f - t : t));
```
Comment (Skylion007) [inline aten/src/ATen/native/mps/kernels/ActivationKernel.metal:349]: No need for any precise mul etc math helpers?

**Review by Skylion007 — APPROVED** [review]

File: torch/testing/_internal/common_modules.py (hunk 1 above)
Comment (malfet) [reply torch/testing/_internal/common_modules.py:4872]: It's not about precise math, but the fact that we test fused op against decomposed one, and fused is more accurate. I'll propose a separate PR to upcast to fp32
